engine/apps/user_management/sync.py

```diff
@@ -132,6 +134,17 @@ def _sync_grafana_incident_plugin(organization: Organization, grafana_api_client
         )
 
 
+def sync_user_from_request(organization: Organization, request: Request) -> User:
```

Comment:
This may be moved somewhere else, and it should eventually create the user from the information in the headers as provided by the backend plugin.